From e648e983ef505c37ea3817ffd9e75a3171e94e65 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 19 Jun 2024 12:53:25 +0300
Subject: [PATCH] [SW-185990] gaudi3: create boot desc v5

Gaudi3 is now using version 5 of boot descriptors, which has the
component version.
This patch sets the component version to a dummy value, and the
actual version will be retrieved in the following patch.

Change-Id: Iec378af310994d355f225ec48a24aa807e9850c7
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/board/habanalabs/gaudi3_arc/post-image-spi.sh b/board/habanalabs/gaudi3_arc/post-image-spi.sh
index 7b491b8..7dce3b8 100755
--- a/board/habanalabs/gaudi3_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-spi.sh
@@ -374,6 +374,29 @@
 	fi
 }
 
+function get_spi_component_ver() {
+	echo "this is a dummy component version"
+}
+
+function print_image_version {
+	local img_name=${1}
+	local img_ver_str=$(get_spi_component_ver $img_name)
+	local max_ver_len=128
+
+	echo -n $img_ver_str | xxd -p
+
+	#pad the remaining space with zeroes
+	local ver_len=${#img_ver_str}
+	local ver_pad=$(($max_ver_len - $ver_len))
+	if [ $ver_len -eq $max_ver_len ]; then
+		return;
+	fi
+
+	for i in $(seq 1 $ver_pad); do
+		printf "%02x" 0
+	done
+}
+
 function print_image_desc {
 	local img_name=$1
 	local img_num=$2
@@ -454,12 +477,15 @@
 	# Prepare Stable and Modifiable image start address low
 	printf "%08x" $start_addr_l | rev_bytes_stdin
 	printf "%08x" $start_addr_l | rev_bytes_stdin
+
+	# Prepare component version
+	print_image_version $img_name
 }
 
-function prep_image_ptrs_common_v4 {
+function prep_image_ptrs_common_v5 {
 	local img_file=$1
 	local img_ptrs_sz=32
-	local img_desc_sz=220
+	local img_desc_sz=348
 	local out_file=$1
 	local KEY_NUM=0
 	local EROM_IMG=$2
@@ -598,7 +624,7 @@
 }
 
 function prep_image_ptrs {
-	prep_image_ptrs_common_v4 $1 $IMAGE_TYPE_SPI
+	prep_image_ptrs_common_v5 $1 $IMAGE_TYPE_SPI
 }
 
 function prep_pldm_preboot_d1_hex {
@@ -613,7 +639,7 @@
 	dd status=none conv=sync bs=16 if=${BINS_DIR}/${EROM_PCIE_BINFILE} of=${BINS_DIR}/${EROM_PCIE_PADDED_BINFILE}
 	dd status=none conv=sync bs=16 if=${BINS_DIR}/${EROM_RECOVERY_BINFILE} of=${BINS_DIR}/${EROM_RECOVERY_PADDED_BINFILE}
 
-	prep_image_ptrs_common_v4 ${STAGE_DIR}/${DEV_PREFIX}-erom-images-pointers.bin.le $IMAGE_TYPE_EROM
+	prep_image_ptrs_common_v5 ${STAGE_DIR}/${DEV_PREFIX}-erom-images-pointers.bin.le $IMAGE_TYPE_EROM
 
 	prep_img_desc_with_crc ${STAGE_DIR}/${EROM_PCIE_BINFILE}
 	prep_img_desc_with_crc ${STAGE_DIR}/${EROM_RECOVERY_BINFILE}
